### PR TITLE
feat: add option for bearer token

### PR DESCRIPTION
**Pull request by mswertz**
Motivation: then we could serve files from Armadillo. Alternatively we could add a type of resource similar to Opal and we can create a PR for that. Please advice if we are on the right track? @marikaris

N.B. not sure about the encoding step.

### CODE REVIEW

File: R/HttpFileResourceGetter.R [hunk 1]
```diff
@@ -49,6 +49,8 @@ HttpFileResourceGetter <- R6::R6Class(
     addHeaders = function(resource) {
       if (!is.null(resource$identity) && nchar(resource$identity)>0 && !is.null(resource$secret) && nchar(resource$secret)>0) {
         httr::add_headers(Authorization = jsonlite::base64_enc(paste0("Basic ", resource$identity, ":", resource$secret)))
+      } else if(!is.null(resource$secret) && nchar(resource$secret)>0) {
+	        httr::add_headers(Authorization = jsonlite::base64_enc(paste0("Bearer",resource$secret)))
```
Comment (ymarcon) [inline R/HttpFileResourceGetter.R:53]: A whitespace is missing between the Bearer and the secret.

**Review by ymarcon — CHANGES_REQUESTED** [review]

File: R/HttpFileResourceGetter.R (hunk 1 above)
Comment (ymarcon) [reply R/HttpFileResourceGetter.R:53]: And it is just the secret that needs to be base64 encoded.

File: R/HttpFileResourceGetter.R [hunk 2]
```diff
@@ -50,7 +50,7 @@ HttpFileResourceGetter <- R6::R6Class(
       if (!is.null(resource$identity) && nchar(resource$identity)>0 && !is.null(resource$secret) && nchar(resource$secret)>0) {
         httr::add_headers(Authorization = jsonlite::base64_enc(paste0("Basic ", resource$identity, ":", resource$secret)))
       } else if(!is.null(resource$secret) && nchar(resource$secret)>0) {
-	        httr::add_headers(Authorization = jsonlite::base64_enc(paste0("Bearer",resource$secret)))
+	httr::add_headers(Authorization = paste0("Bearer", resource$secret))
```
Comment (ymarcon) [inline R/HttpFileResourceGetter.R:53]: Whitespace is still missing after "Bearer", I would expect: `paste0("Bearer ", resource$secret)`

**Review by ymarcon — CHANGES_REQUESTED** [review]

File: R/HttpFileResourceGetter.R (hunk 2 above)
Comment (mswertz) [reply R/HttpFileResourceGetter.R]: Great, done.

**Review by ymarcon — APPROVED** [review]

**Pull request merged**